File: lectures/code/cc_03_01.py
import math

class PointClass:
    def __init__(self, x, y):
        self.x = x
        self.y = y
        
    def dump(self):
        print('Object point@%x x=%f y=%f' % (id(self),self.x,self.y))

    def origin(self):
        return math.sqrt(self.x*self.x+self.y*self.y)
        
pt = PointClass(4.0, 5.0)
PointClass.dump(pt)
print('Origin',PointClass.origin(pt))

pt.dump()
print('Origin',pt.origin())

del(pt)

File: lectures/code/cc_03_01.cs
using System;

public class Point
{
   public double x,y;

   public Point(double cx, double cy) {
       x = cx;
       y = cy;
   }

   public void dump() {
       Console.WriteLine("Point object x={0} y={1}", x, y);
   }

   public double origin() {
        return Math.Sqrt(x*x+y*y);
   }
}

class TestPoint{
    public static void Main(string[] args)
     {
         Point pt = new Point(4.0, 5.0);
         pt.dump();
         Console.WriteLine("Origin {0}", pt.origin());
     }
}

// https://www.programiz.com/csharp-programming/online-compiler/

